src/NHibernate.Test/TestCase.cs

```diff
@@ -303,6 +303,9 @@ protected virtual void DropSchema()
 
 		public static void DropSchema(bool useStdOut, SchemaExport export, ISessionFactoryImplementor sfi, Func<DbConnection> getConnection = null)
 		{
+			using(var optionalConnection = getConnection?.Invoke())
+				export.Drop(useStdOut, true, optionalConnection);
```

Comment:
So my theory Drop commands may also be pooled. So ClearPool should be called after Drop.